src/main/splitting/splitting.py

```diff
@@ -1,117 +1,76 @@
-import os
+import ast
 import logging
 
+import numpy as np
 import pandas as pd
 
+from main.preprocessing.code_tracker_handler import get_ct_language
+from main.util.file_util import get_all_file_system_items, ct_file_condition, get_result_folder, \
+    write_based_on_language, get_name_from_path, get_parent_folder_name
+from main.util.strings_util import does_string_contain_any_of_substrings
 from src.main.util import consts
-from src.main.preprocessing.code_tracker_handler import get_ct_language
-from src.main.splitting.tasks_tests_handler import check_tasks, create_in_and_out_dict
-from src.main.util.file_util import ct_file_condition, get_all_file_system_items, get_name_from_path, \
-    get_parent_folder_name, get_result_folder, write_based_on_language
-
-FRAGMENT = consts.CODE_TRACKER_COLUMN.FRAGMENT.value
-TESTS_RESULTS = consts.CODE_TRACKER_COLUMN.TESTS_RESULTS.value
+from main.util.consts import ISO_ENCODING, FILE_SYSTEM_ITEM, TASK
 
 log = logging.getLogger(consts.LOGGER_NAME)
 
-
-def get_tasks_with_max_rate(tasks: list, test_results: list):
-    max_rate = max(test_results)
-    indices = [i for i, tr in enumerate(test_results) if tr == max_rate]
-    return [tasks[i] for i in indices], max_rate
-
-
-def check_tasks_on_correct_fragments(data: pd.DataFrame, tasks: list, in_and_out_files_dict: dict, file_log_info=''):
-    data[FRAGMENT] = data[FRAGMENT].fillna('')
-    # if run after preprocessing, this value can be taken from 'language' column
-    language = get_ct_language(data)
-    # add unique to logs
-    log.info(f'{file_log_info}, language is {language}, found {str(data.shape[0])} fragments')
-
-    if language is consts.LANGUAGE.NOT_DEFINED.value:
-        data[TESTS_RESULTS] = str([consts.TEST_RESULT.LANGUAGE_NOT_DEFINED.value] * len(tasks))
+TESTS_RESULTS = consts.CODE_TRACKER_COLUMN.TESTS_RESULTS.value
+CHOSEN_TASK = consts.CODE_TRACKER_COLUMN.CHOSEN_TASK.value
+TASK_STATUS = consts.CODE_TRACKER_COLUMN.TASK_STATUS.value
+
+
+def get_solved_task(tests_results: str):
+    tests_results = list(map(int, ast.literal_eval(tests_results)))
+    tasks = consts.TASK.tasks()
+    solved_tasks = [t for i, t in enumerate(tasks) if tests_results[i] == 1]
```

Comment:
I think it would be better if you compare len(tasks) and len(tests_results). Because you use an index i for tasks, but if tests_results contains fewer elements than tasks we should raise an error because it is an incorrect case